spec/forms/providers/application_merits_task/nature_of_urgency_form_spec.rb

```diff
@@ -3,52 +3,105 @@
 module Providers
   module ApplicationMeritsTask
     RSpec.describe NatureOfUrgencyForm do
-      subject(:nature_of_urgency_form) { described_class.new(params) }
+      subject(:form) { described_class.new(params) }
 
       let(:params) do
         {
           nature_of_urgency:,
           hearing_date_set:,
-          hearing_date_1i:,
-          hearing_date_2i:,
-          hearing_date_3i:,
+          hearing_date:,
         }
       end
+
+      let(:a_date) { Time.zone.tomorrow }
       let(:nature_of_urgency) { "Hearing is tomorrow" }
       let(:hearing_date_set) { "true" }
-      let(:hearing_date_1i) { Date.tomorrow.year }
-      let(:hearing_date_2i) { Date.tomorrow.month }
-      let(:hearing_date_3i) { Date.tomorrow.day }
+      let(:hearing_date) { a_date.to_s(:date_picker) }
 
       describe "#valid?" do
         context "when all fields are valid" do
           it "returns true" do
-            expect(nature_of_urgency_form).to be_valid
+            expect(form).to be_valid
           end
         end
 
         context "when hearing_date_set is missing" do
           let(:hearing_date_set) { nil }
 
-          it { expect(nature_of_urgency_form).not_to be_valid }
+          it "is invalid with expected error" do
+            expect(form).to be_invalid
+            expect(form.errors.messages[:hearing_date_set]).to include("Select yes if the hearing date has been set")
+          end
         end
 
         context "when nature_of_urgency is blank" do
           let(:nature_of_urgency) { "" }
 
-          it { expect(nature_of_urgency_form).not_to be_valid }
+          it "is invalid with expected error" do
+            expect(form).to be_invalid
+            expect(form.errors.messages[:nature_of_urgency]).to include("Enter the nature of the urgency")
+          end
         end
+      end
 
-        context "when hearing_date is invalid" do
-          let(:hearing_date_3i) { "xx" }
+      context "when hearing_date_set is false" do
+        let(:hearing_date_set) { "false" }
 
-          it { expect(nature_of_urgency_form).not_to be_valid }
+        context "and hearing_date is not supplied" do
+          let(:hearing_date) { nil }
+
+          it { expect(form).to be_valid }
         end
 
-        context "when hearing_date_set is false" do
-          let(:hearing_date_set) { "false" }
+        # Should it be invalid?/too much
+        context "and hearing_date is supplied" do
+          let(:hearing_date) { a_date.to_s(:date_picker) }
 
-          it { expect(nature_of_urgency_form).to be_valid }
+          it { expect(form).to be_valid }
```

Comment:
I'd expect invalid, could this save data that's not needed/incorrect

Although I suppose the editable application work should only build what's needed?! 🤔